### ISSUE TITLE
SurfaceTrimmer does not keep color info for output

### ISSUE THREAD

**Issue by xiasun** [mkazhdan/PoissonRecon#73]
Follow the [sample usage](http://www.cs.jhu.edu/~misha/Code/PoissonRecon/Version10.01/) to reconstruct surface and trim it.

    ./SSDRecon --in eagle.points.ply --out eagle.screened.color.ply --depth 10 --colors --density
    ./SurfaceTrimmer --in eagle.screened.color.ply --out eagle.screened.color.trimmed.ply --trim 7

The output model of SurfaceTrimmer has no color data. and no color related options is provided by the SurfaceTrimmer program. 

I am not sure is this behavior by design or a bug?
Thank you all.

**Comment by mkazhdan** [comment]
Strange.

Can you type “more eagle.screened.color.ply” and “more eagle.screened.color.trimmed.ply” and check what the lines look like through the line that starts “element face …”?

 

-- Misha

**Comment by mkazhdan** [comment]
That seems very strange indeed.

What machine are you running on (Windows/Linux/Mac)?

What compiler did you use (or are you using the provided Win64 executables)?

**Comment by mkazhdan** [comment]
Very strange. I have compiled under the same version of Ubuntu and gcc but am getting colors in my trimmed result.

**Comment by xiasun** [comment]
Yesterday I tried to debug and found the return value of `hasColor` false with line 388 of SurfaceTrimmer.cpp
```
bool hasColor  = VertexData::ValidPlyReadProperties< 2 >( readFlags + DIMENSION );
```
Then I set it to
```
bool hasColor = true;
```
Color info is correctly processed then. 
But when I try to fix it today, the `hasColor` do have true return value now. I can no longer reproduce the wrong result above myself... It's really strange. 
I think I'd better close this issue now. 
Thank you @mkazhdan !
